app/services/ratio/ratio.rb

```diff
@@ -21,6 +22,13 @@ def relative(currency:)
       JSON.parse(response.body)
     end
 
+    def self.relative_cached(currency:)
```

Comment:
I honestly hate the way we setup the PaymentClient services and think it's horrible for a maintenance standpoint. Sure I understand wanting to stay DRY, but having it be set during the initializer makes our code approach more like spaghetti.

Reply:
Anyone else who jumps in on the project and doesn't realize we setup an initializer arbitrarily would have a hard time figuring out why we're making something so seemingly simple so complex, similarly to how we use Addressable for building out urls rather than using URI.

Reply:
It seems we disagree on the right approach here. I want us to be on the same page and work towards building a code base we both agree is quality.  Can you explain more about your feelings towards maintenance of the services? It would be great to hear some ideas on how you feel like we can make this code less complex, and easier to onboard new developers. If it's easier we can hop on a call and discuss. 😄

Reply:
I think using this client structure for our services makes things a little easier to maintain. 
Instead of having calls to the same services in multiple different places with names that don't associate with the service `PublisherWalletGetter`, `PublisherTransactionGetter` the service client scopes the code under the service name. One problem that I have under the previous approach is that it's hard to answer the question "Does Publishers call this endpoint?". 

In my experience having a single initializer for a client makes it easy to pass configuration in based on the environment. Because there's a single object representing the service it makes it easier to mock service calls out in tests. It's the same approach set forth by most ruby gems which connect to external services. A few good examples are [Yt](https://github.com/brave-intl/publishers/blob/master/config/initializers/yt.rb), [Stripe-ruby](https://stackoverflow.com/questions/35800053/stripe-on-ruby-on-rails-creating-config-initializers-stripe-rb), and [AWS SDK](https://github.com/thepracticaldev/dev.to/blob/master/config/initializers/aws_sdk.rb) gems. It's also nice to have a single object that you can search for (e.g. `PaymentClient`) to see all the instances where the service is being called. 

I've also found some app methodologies that [making no distinction between local and third party services, treating them as attached resources, and loosely coupling them](https://12factor.net/backing-services) is recommend practice. 

We don't have to use addressable. I thought it made things a little easier when dealing with complicated URIs that require multiple substitution, but when you're implementing these methods we can more simply pass a string into any of the `get(url)`, `post(url)`, etc methods.